daemon-protocol/src/main/proto/sonarlint-daemon.proto

```diff
@@ -0,0 +1,120 @@
+// SonarLint, open source software quality management tool.
+// Copyright (C) 2015-2016 SonarSource
+// mailto:contact AT sonarsource DOT com
+//
+// SonarLint is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 3 of the License, or (at your option) any later version.
+//
+// SonarLint is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program; if not, write to the Free Software Foundation,
+// Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+syntax = "proto3";
+
+package sonarlint;
+
+// The java package can be changed without breaking compatibility.
+// it impacts only the generated Java code.
+option java_package = "org.sonarsource.sonarlint.daemon.proto";
+option optimize_for = SPEED;
+
+service ConnectedSonarLint {
+  rpc Register(ConnectedConfiguration) returns (Void);
+  rpc Analyze(ConnectedAnalysisReq) returns (stream Issue);
+  rpc Update(ServerConfig) returns (Void);
+  rpc UpdateModule(ModuleUpdateReq) returns (Void);
+  rpc Status(Void) returns (StorageStatus);
+  rpc Log(Void) returns (stream LogEvent);
+  
+}
+
+service StandaloneSonarLint {
+  rpc Register(StandaloneConfiguration) returns (Void);
+  rpc Analyze(AnalysisReq) returns (stream Issue);
+  rpc Log(Void) returns (stream LogEvent);
+}
+
+message ModuleUpdateReq {
+  ServerConfig server_config = 1;
+  string module_id = 2;
+}
+
+message StorageStatus {
+  Status status = 1;
+  enum Status {
+    UNKNOW = 0;
+    UPDATING = 1;
+    NEVER_UPDATED = 2;
+    NEED_UPDATE = 3;
+    UPDATED = 4;
+  }
+}
+
+message Void {
+}
+
+message LogEvent {
+  string level = 1;
+  string log = 2;
+  bool is_debug = 3;
+}
+
+message ServerConfig {
+  string host_url = 1;
+  string token = 2;
+  string user_agent = 3;
+  string login = 4;
+  string password = 5;
+}
+
+ message Issue {
+   string severity = 1;
```

Comment:
Yep, but using an enum would be more efficient here (string are passed by value and not by reference)